package/src/ssr/NextSSRApolloClient.tsx

```diff
@@ -90,15 +90,36 @@ export class NextSSRApolloClient<
             );
 
             if (!byVariables.has(varJson)) {
+              let resolveFakeQuery: [
+                  (result: FetchResult) => void,
+                  (reason: any) => void
+                ],
+                observable: Observable<FetchResult>,
+                fetchCancelFn: (reason: unknown) => void;
+
+              const cleanup = () => {
+                if (
+                  queryManager["fetchCancelFns"].get(cacheKey) === fetchCancelFn
+                )
+                  queryManager["fetchCancelFns"].delete(cacheKey);
+
+                if (byVariables.get(varJson) === observable)
+                  byVariables.delete(varJson);
```

Comment:
The `byVariables.delete(varJson);` is the relevant part of this PR - the rest is mostly cleanup.